### ISSUE TITLE
CASA flavour syntax error

### ISSUE THREAD

**Issue by senkhxsi** [caracal-pipeline/stimela2#140]
I created a casa.listobs cab (for testing purposes) and when I run a recipe which uses the cab I get an error:

```bash
### running casa --log2term --nologger --nologfile -c                                                                                   
import sys, json;                                                                                                                       
kw=json.loads(sys.argv[-1]);                                                                                                            
def stringify(x):                                                                                                                       
    return str(x) if type(x) is unicode else ( if type(x) is list else x);                                                              
kw={key: stringify(value) for key, value in kw.items()}; listobs(**kw); {"vis": "sgra_eht13ants_00h_15h_tint600s.MS"}                   
# --> CrashReporter initialized.                                                                                                        
# 2023-03-24 10:50:09   INFO    ::casa  optional configuration file config.py not found, continuing CASA startup without it             
# 2023-03-24 10:50:09   INFO    ::casa                                                                                                  
# 2023-03-24 10:50:09   INFO    ::casa  Checking Measures tables in data repository sub-directory                                       
/Applications/CASA.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/casadata/__data__/geodetic         
# 2023-03-24 10:50:09   INFO    ::casa    IERSeop2000 (version date, last date in table (UTC)): 2022/09/01/15:00, 2022/08/02/00:00:00   
# 2023-03-24 10:50:09   INFO    ::casa    IERSeop97 (version date, last date in table (UTC)): 2022/09/01/15:00, 2022/08/02/00:00:00     
# 2023-03-24 10:50:09   INFO    ::casa    IERSpredict (version date, last date in table (UTC)): 2022/09/04/15:00, 2022/12/03/00:00:00   
# 2023-03-24 10:50:09   INFO    ::casa    TAI_UTC (version date, last date in table (UTC)): 2022/08/23/15:00, 2017/01/01/00:00:00       
#                                                                                                                                       
# optional configuration file config.py not found, continuing CASA startup without it                                                   
#                                                                                                                                       
# 0m                                                    
# CASA 6.5.2.26 -- Common Astronomy Software Applications [6.5.2.26]                                                                    
#                                                                                                                                       
/Applications/CASA.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/IPython/core/interactiveshell.py:96
4: UserWarning: Attempting to work in a virtualenv. If you encounter problems, please install IPython inside the virtualenv.            
#   warn(                                                                                                                               
# Telemetry initialized. Telemetry will send anonymized usage statistics to NRAO.                                                       
# You can disable telemetry by adding the following line to the config.py file in your rcdir (e.g. ~/.casa/config.py):                  
# telemetry_enabled = False                                                                                                             
# Traceback (most recent call last):                                                                                                    
#   File                                                                                                                                
"/Applications/CASA.app/Contents/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/casashell/private/init_system.py",
line 238, in __evprop__                                                                                                                 
#     exec(stmt)                                                                                                                        
#   File "<string>", line 6, in <module>                                                                                                
#   File "<string>", line 6, in <dictcomp>                                                                                              
# NameError: name 'stringify' is not defined                                                                                            
# CASA exits with a non-zero status : exception: <class 'NameError'>                                                                    
2023-03-24 12:50:08 STIMELA.test_recipe.list_obs ERROR: step 'test_recipe.list_obs' has failed, aborting the recipe                     
──────────────────────────────────────────────────── detailed error report follows ─────────────────────────────────────────────────────
        ⚠ step 'test_recipe.list_obs' has failed, aborting the recipe                                                                   
        └── error running cab 'listobs'                                                                                                 
            └── listobs returns error code 1 after 0:00:02                                                                               
```

**Comment by sphemakh** [comment]
Pushed a fix as brach issue-140.
1. `unicode` type does not exist in python3 (switched to `bytes`)
2. Rewrote the return statement to use `if isinstance(foo, bar)` instead of `if type(foo) is bar`

https://github.com/caracal-pipeline/stimela/blob/8b545ba2fc5dcc7d6ae4936baa38fd0567bca083/stimela/backends/flavours/casa.py#L80-L92

Checkout the branch and see if it fixes your issue:

```bash
cd <path to where you clonned stimela>
git checkout issue-140
pip install -U ./
```

**Comment by o-smirnov** [comment]
As @SpheMakh  pointed out elsewhere, for CASA6+ tasks, we don't even need CASA flavour as such:

```yml
cabs:
  casa.listobs:
    command: casatasks.listobs
    flavour: python
    inputs:
      vis:
        dtype: MS
        required: True
      overwrite:
        dtype: bool
        default: True
    outputs:
      listfile:
        dtype: File
        default: obsinfo.txt
```

**Comment by sphemakh** [comment]
@o-smirnov coincidentally, there were other errors, which I fixed in PR #150. Please review
